site/source/tutorials/1040_data_driven_widgets/index.md

```diff
@@ -0,0 +1,193 @@
+
+{% instruction 'Update the `ListProperties` interface so that a `value` and an `onInput` callback can be passed into the `List`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/List.ts' lines:9,11-13 %}
+
+The next step in creating an initial filterable list widget is to update its `render` method to define a `TextInput` using the `w` module.
+
+{% instruction 'Update the `List` to render a `TextInput` using the `value` and `onInput` properties.' %}
+
+Click the button below to view the solution.
+
+{% solution showsolution1 %}
+```ts
+protected onInput({ target: { value } }: any) {
+    this.properties.onInput(value);
+}
+
+protected render() {
+    return v('div', [
+        w(TextInput, {
+            value: this.properties.value,
+            onInput: this.onInput,
+            placeholder: 'Filter workers...'
+        })
+    ]);
+}
+```
+{% endsolution %}
+
+Before continuing with the `List` implementation, let’s review and verify the progress so far by adding the current widget to the Biz-E-Bodies application.
+
+{% section %}
+
+## Adding a List widget to the application
+
+{% task 'Add the `List` widget to the application.' %}
+
+We'll add the `List` widget to the existing `Banner` widget so users can filter a list of workers by name when first visiting the application. The end result should look something like this:
+
+<p class="center">![list_widget](../resources/list.png)</p>
+
+{% instruction 'Import the `List` and the `w` module into `Banner.ts`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:2-3 %}
+
+{% instruction 'Try and update `Banner` to render an array of virtual nodes that includes the existing `<h1>` element as well as a `List`.' %}
+
+{% solution showsolution2 %}
+```ts
+protected render() {
+    return [
+        v('h1', { title: 'I am a title!' }, [ 'Welcome To Biz-E-Bodies' ]),
+        v('label', ['Find a worker:']),
+        w(List, {
+            onInput: (value: string) => null,
+            value: ''
+        })
+    ];
+}
+```
+{% endsolution %}
+
+{% instruction 'Run the application with `dojo build --watch` and navigate to [`http://localhost:9999`](http://localhost:9999).' %}
+
+A solid foundation for a filterable list widget using a Dojo 2 `TextInput` widget has been created and added to the existing `Banner`. Up to this point, the widget has no knowledge of an external data source.
+
+{% section %}
+
+## Connecting the List to data
+
+{% task 'Connect the `List` to data using its `properties`.' %}
+
+Traditional widget frameworks like Dojo 1 required developers to tightly couple widget instances to data store instances. For example, it was common for Dojo 1 widget code to expect a `store` property to exist on a widget instance, and to further expect that store to expose a Dojo-specific data store API. While effective, this method of explicitly connecting widgets to data stores is brittle and forces widgets to have knowledge of a store implementation.
```

Comment:
"is brittle and forces widgets to have knowledge of a store implementation" I'm not sure an average user at this point is going to understand why this is brittle or why that knowledge is a bad thing. I'm thinking that maybe it makes sense to explain this differentiation after you've shown the benefits of Dojo 2 stores, or maybe simply omit this last sentence altogether?

Reply:
I'm fine with leaving this out or revisiting it later in the tutorial, but the more I think about it, the more I feel like this part of the last sentence adds enough clarity around why the Dojo 1 approach is brittle: "forces widgets to have knowledge of a store implementation." Maybe I should change this to say something like: "forces widgets to call specific methods on stores"?

Reply:
I think the suggested change in your last sentence helps clarify.